prov/sockets/src/sock_msg.c

```diff
@@ -59,6 +59,18 @@
 #define SOCK_LOG_INFO(...) _SOCK_LOG_INFO(FI_LOG_EP_DATA, __VA_ARGS__)
 #define SOCK_LOG_ERROR(...) _SOCK_LOG_ERROR(FI_LOG_EP_DATA, __VA_ARGS__)
 
+static int sock_drop_packet(struct sock_ep *sock_ep)
+{	
+#if ENABLE_DEBUG	
+	if(sock_ep->ep_type == FI_EP_DGRAM && sock_dgram_drop_rate > 0) {
+	sock_ep->domain->fab->num_send_msg;
+		if(!(sock_ep->domain->fab->num_send_msg % sock_dgram_drop_rate))
+			return 1;
+	}
+#endif
+	return 0;
+}
+
 static ssize_t sock_ep_recvmsg(struct fid_ep *ep, const struct fi_msg *msg, 
```

Comment:
I think it might be better to move this into the header file, as Sean indicated.